kratos/python_interface/python_registry.py

```diff
@@ -0,0 +1,370 @@
+        return py_keys, cpp_keys
+
+    def values(self, Name):
+        # Check if Name registry level is registered and if it is a value
+        if self.HasItem(Name):
+            if not self.HasItems(Name):
+                err_mgs = f"Asking for the keys of '{Name}'. '{Name}' item has no subitems."
+                raise Exception(err_mgs)
+        else:
+            err_msg = f"Asking for the keys of '{Name}' non-registered item."
+            raise Exception(err_msg)
+
+        # Get the c++ registry item values
+        cpp_values = None
+        if self.HasItems(Name, RegistryContext.CPP):
+            cpp_values = self.__InternalGetCppItem(Name).values()
+        # Get the Python registry item values
+        # Note that this is a view object of the Python dictionary values
+        py_values = None
+        if self.HasItems(Name, RegistryContext.PYTHON):
+            py_values = self.__InternalGetPythonItem(Name).values()
+
+        # Return the c++ and Python registry values in a tuple
+        # Note that we return None if the item was not present in one of the registries
+        return py_values, cpp_values
+
+    def items(self, Name):
+        # Check if Name registry level is registered and if it is a value
+        if self.HasItem(Name):
+            if not self.HasItems(Name):
+                err_mgs = f"Asking for the keys of '{Name}'. '{Name}' item has no subitems."
+                raise Exception(err_mgs)
+        else:
+            err_msg = f"Asking for the keys of '{Name}' non-registered item."
+            raise Exception(err_msg)
+
+        # Get the keys and values
+        py_keys, cpp_keys = self.keys(Name)
+        py_values, cpp_values = self.values(Name)
+
+        # Loop the keys and values to return the items
+        py_items = None
+        if py_keys is not None:
+            py_items = []
+            for py_key, py_value in zip(py_keys, py_values):
+                py_items.append((py_key, py_value))
+
+        cpp_items = None
+        if cpp_keys is not None:
+            cpp_items = []
+            for cpp_key, cpp_value in zip(cpp_keys, cpp_values):
+                cpp_items.append((cpp_key, cpp_value))
+
+        # Return the c++ and Python registry items in a tuple
+        return py_items, cpp_items
+
+    def NumberOfItems(self, Name):
+        '''Returns the total number of items in the registry
+        '''
+        py_keys, cpp_keys = self.keys(Name)
+        total_items = 0
+        if py_keys is not None:
+            total_items += len(py_keys)
+        if cpp_keys is not None:
+            total_items += len(cpp_keys)
+        return total_items
+
+    def NumberOfPythonItems(self, Name):
+        '''Returns the number of Python registered items
+        '''
+        py_keys, _ = self.keys(Name)
+        total_items = 0
+        if py_keys is not None:
+            total_items += len(py_keys)
+        return total_items
+
+    def NumberOfCppItems(self, Name):
```

Comment:
Todo: We may add this method in CPP to be more efficient. (In a separate PR)

Reply:
Agree. I'll add a TODO comment as reminder.